samples/parselib/src/main/cpp/wav/WavStreamReader.cpp

```diff
@@ -110,48 +110,203 @@ void WavStreamReader::positionToAudio() {
     }
 }
 
-int WavStreamReader::getDataFloat(float *buff, int numFrames) {
-    // __android_log_print(ANDROID_LOG_INFO, TAG, "getData(%d)", numFrames);
+/**
+ * Read and convert samples in PCM8 format to float
+ */
+int WavStreamReader::getDataFloat_PCM8(float *buff, int numFrames) {
+    int numChans = mFmtChunk->mNumChannels;
 
-    if (mDataChunk == 0 || mFmtChunk == 0) {
-        return 0;
+    int buffOffset = 0;
+    int totalFramesRead = 0;
+
+    const static int SAMPLE_SIZE = sizeof(u_int8_t);
+    const static float SAMPLE_FULLSCALE = (float)0x7F;
+    const static int CONVERT_BUFFER_FRAMES = 128;
+
+    u_int8_t readBuff[CONVERT_BUFFER_FRAMES * numChans];
+    int framesLeft = numFrames;
+    while (framesLeft > 0) {
+        int framesThisRead = std::min(framesLeft, CONVERT_BUFFER_FRAMES);
+        //__android_log_print(ANDROID_LOG_INFO, TAG, "read(%d)", framesThisRead);
+        int numFramesRead =
+                mStream->read(readBuff, framesThisRead *  SAMPLE_SIZE * numChans) /
+                (SAMPLE_SIZE * numChans);
+        totalFramesRead += numFramesRead;
+
+        // Convert & Scale
+        for (int offset = 0; offset < numFramesRead * numChans; offset++) {
+            // PCM8 is unsigned, so we need to make it signed before scaling/converting
+            buff[buffOffset++] = ((float) readBuff[offset] - SAMPLE_FULLSCALE)
+                    / (float) SAMPLE_FULLSCALE;

```

Comment:
It is faster to use to an inverse scaler and multiply instead of dividing.  Surprisingly, compilers often do not make this optimization.

    constexpr float kInverseScaler = 1.0 / 0x80;

Reply:
Done.